sycl/doc/extensions/proposed/sycl_ext_intel_fpga_datapath.asciidoc

```diff
@@ -212,24 +264,39 @@ Returns a reference to the underlying `T`.
 
 Below are some non-obvious interactions between `fpga_datapath` and other SYCL extensions. 
 
-* The `fpga_datapath` and `fpga_mem` are tempate classes that have conflicting requirements on how a variable should be implemented, on or off the datapath. In situations where an object of one of these classes is templated on another, the most specific template will used by the compiler, and the rest will be ignored.
+* The `fpga_datapath` and `fpga_mem` are tempate classes that have conflicting requirements on how a variable should be implemented, on or off the datapath. In situations where an object of one of these classes is templated on another, the less specific template will used by the compiler, and the rest will be ignored.
 
 [source,c++]
 ----
 namespace intelexp = sycl::ext::intel::experimental;
 
-// 10 integers will be implemented on the datapath
-// `fpga_mem` is ignored.
+// 10 integers will be implemented off the datapath, in memory
+// `fpga_datapath` is ignored.
 intelexp::fpga_mem<intelexp::fpga_datapath<int[10]>> dm1;
 
+----
+
+* `fpga_mem` and `fpga_datapath` objects can be member variables of a struct. If the struct type itself is used as a template argument to `fpga_mem` or `fpga_datapath` any hints provided from the member variables are ignored. 
+
+[source,c++]
+----
+namespace intelexp = sycl::ext::intel::experimental;
+
 class foo {
   intelexp::fpga_mem<int[5]> a;
   intelexp::fpga_datapath<int[4]> b;
-  int[3] c;
 }
 
-// 5 integers of `a` will be implemented in a memory off the datapath
-// 4 integers of `b` will be implemented on the datapath
-// 3 integers of `c` will be implemented in a memory off the datapath
+// `var` will be split into two variables
+// `var.a` will be implemented in memory, off the datapath
+// `var.b` will be implemented in the datapath
+foo var;
+
+// Hints applied on member variables ignored
+// `dm1` implemented in memory, off the datapath
 intelexp::fpga_mem<foo> dm1;
+
+// Hints applied on member variables ignored
+// `dm2` implemented in the datapath
```

Comment:
Mostly for my own education, what is the differences between "on", "off" and "in" the datapath?

Reply:
On the datapath, means that the value of the variable is propagated through the program via wires in the HW, or sometimes registers.
Off the datapath, or in other words "in memory", HW is dedicated to build memory where the data of the variable is stored. Accesses to the variable require load/store operations to execute. 

Ultimately whether a variable is implemented in the datapath or off the datapath, is a trade off between latency of access (variables in the datapath would be 0 cycle accesses latency), and the practical size of the data (building a memory is much more resource efficient than having a wire per bit). 

There are a few other nuances, but that is the gist.

Reply:
Here it says "in the datapath". I assume that is synonymous to "on the datapath"?

Reply:
Yes that is a my bad